tree/ntuple/v7/src/RField.cxx

```diff
@@ -146,40 +146,42 @@ std::tuple<std::string, std::vector<size_t>> ParseArrayType(std::string_view typ
    return std::make_tuple(std::string{typeName}, sizeVec);
 }
 
-/// Returns the normalized resolved type and the normalized typedef before typedef resolution. If the given type
-/// is not a typedef, the second element of the returned pair is the empty string.
-std::pair<std::string, std::string> GetNormalizedType(const std::string &typeName) {
-   std::string normalizedType(TClassEdit::CleanType(typeName.c_str(), /*mode=*/2));
+/// Return the canonical name of a type, resolving typedefs to their underlying types if needed
+std::string GetCanonicalTypeName(const std::string &typeName)
+{
    // The following types are asummed to be canonical names; thus, do not perform `typedef` resolution on those
-   if (normalizedType == "ROOT::Experimental::ClusterSize_t")
-      return std::make_pair(normalizedType, "");
-   // We want the template parameter to stay in "std::uint32_t" / "std::uint64_t" form
-   if (normalizedType.substr(0, 39) == "ROOT::Experimental::RNTupleCardinality<")
-      return std::make_pair(normalizedType, "");
-
-   std::string resolvedType = TClassEdit::ResolveTypedef(normalizedType.c_str());
-   if (resolvedType == normalizedType)
-      normalizedType = "";
-   auto translatedType = typeTranslationMap.find(resolvedType);
-   if (translatedType != typeTranslationMap.end())
-      resolvedType = translatedType->second;
-
-   if (resolvedType.substr(0, 7) == "vector<")
-      resolvedType = "std::" + resolvedType;
-   if (resolvedType.substr(0, 6) == "array<")
-      resolvedType = "std::" + resolvedType;
-   if (resolvedType.substr(0, 8) == "variant<")
-      resolvedType = "std::" + resolvedType;
-   if (resolvedType.substr(0, 5) == "pair<")
-      resolvedType = "std::" + resolvedType;
-   if (resolvedType.substr(0, 6) == "tuple<")
-      resolvedType = "std::" + resolvedType;
-   if (resolvedType.substr(0, 7) == "bitset<")
-      resolvedType = "std::" + resolvedType;
-   if (resolvedType.substr(0, 11) == "unique_ptr<")
-      resolvedType = "std::" + resolvedType;
-
-   return std::make_pair(resolvedType, normalizedType);
+   if (typeName == "ROOT::Experimental::ClusterSize_t" || typeName.substr(0, 5) == "std::" ||
+       typeName.substr(0, 39) == "ROOT::Experimental::RNTupleCardinality<")
+      return typeName;
+
+   return TClassEdit::ResolveTypedef(typeName.c_str());
+}
+
+/// Applies type name normalization rules that lead to the final name used to create a RField, e.g. transforms
+/// `unsigned int` to `std::uint32_t` or `const vector<T>` to `std::vector<T>`
```

Comment:
This (pre-existing) nomenclature introduces a confusion for the meaning of Normalized.  For Core/Meta, normalized means to remove the typedef and the `std::` while it seems here that it is the opposite.   'Canonical' is also overloaded as it is used by Clang for a (slightly) different meaning.   I recommend that we more clearly distinguish those.